x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/esql/action/ColumnInfo.java

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.esql.action;
+
+public interface ColumnInfo {
```

Comment:
ColumnInfo is a record in the ES|QL module. The reason I separated this out is that there is a bunch of xcontent parsing related "stuff" in it. It can be refactored out, allowing to declare this type a record here. I'll do that separately and merge in here when done.